ultrack/core/export/trackmate.py

```diff
@@ -26,14 +26,68 @@ def tracks_layer_to_trackmate(
+    --------
+    >>> tracks_df = pd.DataFrame(
+    ...     [[1,0,12.0,49.0,49.0,1000001,-1,-1],
+    ...     [1,1,12.0,49.0,32.0,2000001,-1,1000001],
+    ...     [2,1,12.0,49.0,66.0,2000002,-1,1000001]],
+    ...     columns=['track_id','t','z','y','x','id','parent_track_id','parent_id']
+    ... )
+    >>> print(tracks_df)
+       track_id  t     z     y     x       id  parent_track_id  parent_id
+    0         1  0  12.0  49.0  49.0  1000001               -1         -1
+    1         1  1  12.0  49.0  32.0  2000001               -1    1000001
+    2         2  1  12.0  49.0  66.0  2000002               -1    1000001
+    >>> tracks_layer_to_trackmate(tracks_df)
+    <?xml version="1.0" ?>
+    <TrackMate version="7.11.1">
+        <Model spatialunits="pixels" timeunits="frames">
+            <AllTracks>
+                <Track TRACK_ID="1" NUMBER_SPOTS="2" NUMBER_GAPS="0" TRACK_START="0" TRACK_STOP="1" name="Track_1">
+                    <Edge SPOT_SOURCE_ID="0" SPOT_TARGET_ID="1" EDGE_TIME="0.5"/>
+                </Track>
+                <Track TRACK_ID="2" NUMBER_SPOTS="1" NUMBER_GAPS="0" TRACK_START="1" TRACK_STOP="1" name="Track_2">
+                    <Edge SPOT_SOURCE_ID="0" SPOT_TARGET_ID="2" EDGE_TIME="0.5"/>
+                </Track>
+            </AllTracks>
+            <FilteredTracks>
+                <TrackID TRACK_ID="1"/>
+                <TrackID TRACK_ID="2"/>
+            </FilteredTracks>
+            <AllSpots>
+                <SpotsInFrame frame="0">
+                    <Spot ID="0" QUALITY="1.0" VISIBILITY="1" name="0" FRAME="0" RADIUS="5.0" POSITION_X="49.0" POSITION_Y="49.0" POSITION_Z="12.0"/>
+                </SpotsInFrame>
+                <SpotsInFrame frame="1">
+                    <Spot ID="1" QUALITY="1.0" VISIBILITY="1" name="1" FRAME="1" RADIUS="5.0" POSITION_X="32.0" POSITION_Y="49.0" POSITION_Z="12.0"/>
+                    <Spot ID="2" QUALITY="1.0" VISIBILITY="1" name="2" FRAME="1" RADIUS="5.0" POSITION_X="66.0" POSITION_Y="49.0" POSITION_Z="12.0"/>
+                </SpotsInFrame>
+            </AllSpots>
+            <FeatureDeclarations>
+                ...
+            </FeatureDeclarations>
+        </Model>
+        <Settings>
+            <InitialSpotFilter feature="QUALITY" value="0.0" isabove="true"/>
+            <SpotFilterCollection/>
+            <TrackFilterCollection/>
+            <ImageData filename="None" folder="None" width="0" height="0" depth="0" nslices="1" nframes="2" pixelwidth="1.0" pixelheight="1.0" voxeldepth="1.0" timeinterval="1.0"/>
+        </Settings>
+    </TrackMate>
     """
+    if not pd.api.types.is_integer_dtype(tracks_df.index.dtype):
+        raise ValueError("The DataFrame index will be TrackMate spot IDs and must be an integer type.")
     tracks_df["id"] = tracks_df["id"].astype(int)
+    if not tracks_df["id"].is_unique:
+        raise ValueError("The 'id' column must be unique.")
     tracks_df["parent_id"] = tracks_df["parent_id"].astype(int)
```

Comment:
I think this bug would be fixed by adding
```python
tracks_df = tracks_df.set_index("id")
```
here

Reply:
Hey @JoOkuma, very elegant fix. I think you are right. Let me check it on my dataset.